lib/instrumentation/modules/cassandra-driver.js

```diff
@@ -0,0 +1,186 @@
+'use strict'
+
+const semver = require('semver')
+const sqlSummary = require('sql-summary')
+
+const shimmer = require('../shimmer')
+
+module.exports = function (cassandra, agent, version, enabled) {
+  if (!enabled) return cassandra
+  if (!semver.satisfies(version, '^3.0.0')) {
+    agent.logger.debug('cassandra-driver version %s not supported - aborting...', version)
+    return cassandra
+  }
+
+  if (cassandra.Client) {
+    shimmer.wrap(cassandra.Client.prototype, 'connect', wrapConnect)
+    shimmer.wrap(cassandra.Client.prototype, 'execute', wrapExecute)
+    shimmer.wrap(cassandra.Client.prototype, 'eachRow', wrapEachRow)
+    shimmer.wrap(cassandra.Client.prototype, 'batch', wrapBatch)
+  }
+
+  return cassandra
+
+  function wrapConnect (original) {
+    return function wrappedConnect (callback) {
+      const span = agent.buildSpan()
+      if (!span) {
+        return original.apply(this, arguments)
+      }
+
+      span.type = 'db.cassandra.connect'
+      span.start()
+
+      function resolve () {
+        span.end()
+      }
+
+      // Wrap the callback
+      const ret = original.call(this, wrapCallback(callback))
+
+      if (typeof callback !== 'function') {
+        if (typeof ret.then === 'function') {
+          ret.then(resolve, resolve)
+        } else {
+          agent.logger.error('unable to identify span exit point for cassandra-driver')
+        }
+      }
+
+      return ret
+
+      function wrapCallback (cb) {
+        if (typeof cb !== 'function') return cb
+        return function wrappedCallback () {
+          resolve()
+          return cb.apply(this, arguments)
+        }
+      }
+    }
+  }
+
+  function toQueryString (query) {
+    return query.query
+  }
+
+  function wrapBatch (original) {
+    return function wrappedBatch (queries, options, callback) {
+      const span = agent.buildSpan()
+      if (!span) {
+        return original.apply(this, arguments)
+      }
+
+      const queryStrings = queries.map(toQueryString)
+      const querySummaries = queryStrings.map(sqlSummary)
+      const query = queryStrings.join(';\n')
+      const querySummary = querySummaries.join('; ')
```

Comment:
I'm worried this might become a really long name in case there's a lot of queries.

Possible solutions:

1. Only show the first query summary. This is how we do it for SQL query batches in mysql and postgresql today. But this is actually just out of laziness and should probably be reconsidered anyway
1. Make sure all query summaries are unique, e.g. `querySummaries.filter(unique).join('; ')`
1. Don't show any summary at all and just name the span `Batch Query`

cc @beniwohli

Reply:
Yeah, I think something like just `BATCH` is better for the name than joining all the queries. It gets long quickly, even summarized.

Reply:
Any reason why you want to uppercase it?

Reply:
Not really. Sometimes people capitalize keywords in SQL, and I was thinking of it kind of like that, but whatever works. 🤷‍♂️

Reply:
Makes sense. Maybe just `batch query`?

Reply:
We have previously prefixed certain span names with the module that created them, e.g. Elasticsearch span names are prefixed `Elasticsearch: `. For SQL queries we don't normally do this as it's usually pretty obvious where they are coming from, but just as with Elasticsearch we might consider prefixing Cassandra spans as well. If so it would be something like `Cassandra: Batch query`. What do you think?

Reply:
fixed